src/Servers.jl

```diff
@@ -268,140 +181,128 @@ e.g.
+            conn = Connection(io)
+            conn.host, conn.port = server.hostname, server.hostport
+            let io=io, count=count
+                @async try
+                    verbose && @info "Accept ($count):  $conn"
+                    handle_connection(h, conn, reuse_limit, readtimeout)
+                    verbose && @info "Closed ($count):  $conn"
+                catch e
+                    if e isa Base.IOError && e.code == -54
+                        verbose && @warn "connection reset by peer (ECONNRESET)"
+                    else
+                        @error exception=(e, stacktrace(catch_backtrace()))
                     end
-                end
-            catch e
-                if e isa Base.IOError
-                    @warn "$e"
-                    break
-                else
-                    rethrow(e)
+                finally
+                    connection_count[] -= 1
+                    close(io)
+                    verbose && @info "Closed ($count):  $conn"
                 end
             end
-            id += 1
-        end
-    catch e
-        if e isa InterruptException
-            @warn "Interrupted: listen($hostname)"
-        else
-            rethrow(e)
+        catch e
+            if e isa InterruptException
+                @warn "Interrupted: listen($server)"
+                close(server)
+                break
+            else
+                rethrow(e)
+            end
         end
-    finally
-        close(tcpserver)
+        count += 1
     end
-
     return
 end
 
-
 """
-Start a timeout monitor task to close the `Connection` if it is inactive.
-Create a `Transaction` object for each HTTP Request received.
+Connection handler: starts an async readtimeout thread if needed, then creates
+Transactions to be handled as long as the Connection stays open. Only reuse_limit + 1
+# of Transactions will be allowed during the lifetime of the Connection.
 """
-function handle_connection(f::Function, c::Connection;
-                           reuse_limit::Int=nolimit,
-                           readtimeout::Int=0, kw...)
-
+function handle_connection(h, c::Connection, reuse_limit, readtimeout)
     wait_for_timeout = Ref{Bool}(true)
-    if readtimeout > 0
-        @async while wait_for_timeout[]
-            @show inactiveseconds(c)
-            if inactiveseconds(c) > readtimeout
-                @warn "Timeout: $c"
-                writeheaders(c.io, Response(408, ["Connection" => "close"]))
-                close(c)
-                break
-            end
-            sleep(8 + rand() * 4)
-        end
-    end
-
+    readtimeout > 0 && check_readtimeout(c, readtimeout, wait_for_timeout)
     try
         count = 0
         while isopen(c)
-            io = Transaction(c)
-            handle_transaction(f, io; final_transaction=(count == reuse_limit), kw...)
+            handle_transaction(h, Transaction(c); final_transaction=count == reuse_limit)
```

Comment:
If this is now supposed to be "no change" it should just revert back to:
```julia
io = Transaction(c)
handle_transaction(f, io; final_transaction=(count == reuse_limit))
```

Reply:
@samoconnor you're being exhausting; this is literally a super nitpicky comment; let's please move past things like this and focus on meaningful changes. If you want to come back in and reformat things post-merge, that's fine, but you're creating an extremely antagonistic review process here; the real changes are getting lost in these much less meaningful formatting comments. This is open source; we need to be welcoming, encouraging, and flexible with contributions. I've mentioned elsewhere, if you have specific formatting styles you'd like HTTP.jl to commit to, then let's open an issue, discuss, and commit/document. I'm literally getting to the point of not wanting to contribute to the package anymore w/ this PR. Please, just be a little more flexible here?